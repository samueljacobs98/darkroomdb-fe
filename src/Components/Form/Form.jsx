import "./Form.scss";

const Form = () => {
  return (
    <form className="form">
      <label htmlFor="filmName">Film Name*</label>
      <input type="text" id="filmName" name="filmName" required />
      <label htmlFor="rating">Rating</label>
      <input type="number" id="rating" name="rating" min="0" max="5" />
      <label htmlFor="information">Information*</label>
      <textarea id="information" name="information" rows="4" required />
      <label htmlFor="price">Price (Â£) per role*</label>
      <input
        type="number"
        step="0.10"
        placeholder="0.00"
        id="price"
        name="price"
        required
      />
      <label htmlFor="formats">
        Available Formats*
        <br />
        (separate by commas)
      </label>
      <input type="text" id="formats" name="formats" required />
      <label htmlFor="formats">Style*</label>
      <select type="text" id="formats" name="formats" required>
        <option value="black and white">Black and white</option>
        <option value="black and white">Colour</option>
        <option value="black and white">Sepia</option>
      </select>
      <button type="submit">Submit</button>
      <p>* required</p>
    </form>
  );
};

export default Form;

// Name
// Rating
// Info
// Price
// Available Formats
// Style -> dropdown
