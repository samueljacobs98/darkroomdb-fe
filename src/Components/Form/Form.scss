@use "../../styles/variables" as *;

.form {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: rgba($colour-primary, 0.2);
  padding: $gap;
  padding-bottom: calc($gap * 1.5);
  text-align: center;
  position: relative;

  & > label,
  & > button {
    font-size: $fs-secondary;
  }

  & > input,
  & > textarea,
  & > select {
    margin-top: calc($gap / 3);
    margin-bottom: calc($gap / 2);
    color: $colour-primary;
    font-size: $fs-secondary;
    width: 100%;
  }

  & > button {
    background-color: rgba($colour-tertiary, 0.5);
    border: none;
    padding: 0.5rem;

    &:hover {
      cursor: pointer;
    }
  }

  & > p {
    font-size: 0.8rem;
    position: absolute;
    bottom: calc($gap / 2);
    right: calc($gap / 2);
  }
}
