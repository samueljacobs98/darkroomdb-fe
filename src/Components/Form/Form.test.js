import { render } from "@testing-library/react";
import Form from "./Form";

it("should match component snapshot", () => {
  const { container } = render(<Form />);

  expect(container).toMatchSnapshot();
});
