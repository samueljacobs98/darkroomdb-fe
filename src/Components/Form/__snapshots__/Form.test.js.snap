// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match component snapshot 1`] = `
<div>
  <form
    class="form"
  >
    <label
      for="filmName"
    >
      Film Name*
    </label>
    <input
      id="filmName"
      name="filmName"
      required=""
      type="text"
    />
    <label
      for="rating"
    >
      Rating
    </label>
    <input
      id="rating"
      max="5"
      min="0"
      name="rating"
      type="number"
    />
    <label
      for="iso"
    >
      ISO*
    </label>
    <input
      id="iso"
      min="0"
      name="iso"
      required=""
      type="number"
    />
    <label
      for="information"
    >
      Information*
      <br />
      (max. 250 characters)
    </label>
    <textarea
      id="information"
      maxlength="250"
      name="information"
      required=""
      rows="4"
    />
    <label
      for="formats"
    >
      Available Formats*
      <br />
      (separate by commas)
    </label>
    <input
      id="formats"
      name="formats"
      required=""
      type="text"
    />
    <label
      for="style"
    >
      Style*
    </label>
    <select
      id="style"
      name="style"
      required=""
      type="text"
    >
      <option
        value="black and white"
      >
        Black and white
      </option>
      <option
        value="colour"
      >
        Colour
      </option>
      <option
        value="sepia"
      >
        Sepia
      </option>
    </select>
    <button
      type="submit"
    >
      Submit
    </button>
    <p>
      * required
    </p>
  </form>
</div>
`;
